### PR TITLE
Tutorial generic vcf

### PR DESCRIPTION
**Pull request by agladstein**
Add a tutorial for python API generic model and output a vcf.

I was originally intending to also show how to output a tped file. But, on second thought, doesn't seem that useful to output a tped once you already show how to get a vcf file. PLINK takes vcf and can convert to a tped.

closes https://github.com/popgensims/stdpopsim/issues/173

### CODE REVIEW

**Comment by andrewkern** [comment]
this looks really good to me @agladstein. can you rebase this against master?

**Comment by agladstein** [comment]
Do you want me to squash the commits, or is it okay to leave those two?

**Comment by andrewkern** [comment]
might as well squash to make it clean. thanks!

File: docs/tutorial.rst [hunk 1]
```diff
@@ -118,129 +118,118 @@ The Python interface
...
+1. Import the necessary packages
 
-.. Recombination maps will be downloaded on demand and cached in a
-.. platform-appropriate user cache directory (e.g., ``$HOME/.cache/stdpopsim`` on
-.. Linux). In this example we didn't specify which recombination map we want, and so the
-.. API will use the default. We can also ask for specific maps, if we want:
+.. code-block:: python
 
-.. .. code-block:: python
+    >>> import msprime
```
Comment (jeromekelleher) [inline docs/tutorial.rst]: We don't actually need msprime here, do we?

File: docs/tutorial.rst [hunk 2]
```diff
@@ -118,129 +118,118 @@ The Python interface
...
+population size for one population over time.
 
+.. code-block:: python
 
+    >>> model = stdpopsim.PiecewiseConstantSize(species.population_size)
 
-.. _sec_tutorial_generic_models:
+5. Set the number of samples and set the simulation engine.
+In this case we will simulate 10 samples and use the default simulator,
+which is `msprime`. But, you can go crazy with the sample size!
```
Comment (jeromekelleher) [inline docs/tutorial.rst]: :laughing:

File: docs/tutorial.rst [hunk 3]
```diff
@@ -118,129 +118,118 @@ The Python interface
...
-The previous sections showed how to simulate population genetic models that have
-been published for particular species. It is also sometimes useful to simulate
-more generic models. We can still use the information for particular species to
-do this; the only difference is that we instantiate the models directly rather
-than retrieving them from the :ref:`catalog <sec_catalog>`.
+    >>> samples = model.get_samples(10)
+    >>> engine = stdpopsim.get_default_engine()
 
+6. Simulate the model with the contig length and number of samples we defined above.
+We capture the simulation results in a `tree sequence object <https://tskit.readthedocs.io/en/stable/python-api.html#tskit.TreeSequence>`_.
```
Comment (jeromekelleher) [inline docs/tutorial.rst]: Can link automatically using `:class:`tskit.TreeSequence``

Comment (jeromekelleher) [reply docs/tutorial.rst]: Ugh, hard to do quoteing of backticks in markdown! you'll get the idea.

File: docs/tutorial.rst [hunk 4]
```diff
@@ -118,129 +118,118 @@ The Python interface
...
 
-    species = stdpopsim.get_species("homsap")
-    contig = species.get_contig("chr22", length_multiplier=0.1)
-    model = stdpopsim.PiecewiseConstantSize(species.population_size)
-    samples = model.get_samples(10)
-    engine = stdpopsim.get_default_engine()
-    ts = engine.simulate(model, contig, samples)
+    >> ts = engine.simulate(model, contig, samples)
 
+7. We can now do some simple checks that our simulation worked with ``tskit``.
```
Comment (jeromekelleher) [inline docs/tutorial.rst]: Maybe link to the tskit docs here: tskit.readthedocs.io

File: docs/tutorial.rst [hunk 5]
```diff
@@ -118,129 +118,118 @@ The Python interface
...
+As expected, there are 10 samples in one population. We can also see that 6197 mutations
+and 6863 trees were simulated (since we are not using a seed here, the number of mutations
+and trees will be slightly different for you). Try running the simulation again, and notice
+that the number of samples and populations stays the same, while the number of mutations
+and trees changes.
+
+8. In addition to working directly with the simulated tree squence, we can also output
+other common formats used for population genetics analyses.
+We can use ``tskit`` to convert the tree sequence to a vcf file called "foo.vcf".
+See the `tskit documentation
```
Comment (jeromekelleher) [inline docs/tutorial.rst]: Again, can link automatically.

File: docs/tutorial.rst (hunk 1 above)
Comment (agladstein) [reply docs/tutorial.rst]: You are right. Good catch.

File: docs/tutorial.rst (hunk 5 above)
Comment (agladstein) [reply docs/tutorial.rst]: Haven't figured out how to get this to link.
I tried
```
:class:`tskit.TreeSequence.write_vcf`
```

File: docs/tutorial.rst (hunk 4 above)
Comment (agladstein) [reply docs/tutorial.rst]: Can I make a direct link?

I was going to do:
```
``tskit``
(https://tskit.readthedocs.io)
```

File: docs/tutorial.rst (hunk 5 above)
Comment (jeromekelleher) [reply docs/tutorial.rst]: This is a method, so ``:meth:``

File: docs/tutorial.rst (hunk 4 above)
Comment (jeromekelleher) [reply docs/tutorial.rst]: It's something like `tskit <https://tskit.readthedocs.io>`_

Comment (jeromekelleher) [reply docs/tutorial.rst]: Gah, trying to do RST syntax in markdown is a pain! See the second example here: http://docutils.sourceforge.net/docs/user/rst/quickref.html#hyperlink-targets

File: docs/tutorial.rst (hunk 5 above)
Comment (agladstein) [reply docs/tutorial.rst]: thanks

File: docs/tutorial.rst (hunk 4 above)
Comment (agladstein) [reply docs/tutorial.rst]: ok cool.

and just make a big code block, then you can have the RST syntax.

**Comment by agladstein** [comment]
Okay, I added in @jeromekelleher's comments.

**Comment by agladstein** [comment]
I just noticed that the section header was the wrong size. I fixed it and made it more informative.

**Comment by jeromekelleher** [comment]
Great, thanks @agladstein. Merging.

**Pull request merged**